File: GXpert/GXpert.Web/Modules/Exams/Exam/ExamForm.cs

using Serenity.ComponentModel;
using System;

namespace GXpert.Exams.Forms;

[FormScript("Exams.Exam")]
[BasedOnRow(typeof(ExamRow), CheckNames = true)]
public class ExamForm
{
    public string Title { get; set; }
    [HalfWidth]
    public short EExamType { get; set; }
    [HalfWidth]
    public short EExamState { get; set; }
    [HalfWidth]
    public string QuestionPaperPath { get; set; }
    [HalfWidth]
    public string ModelAnswerPaperPath { get; set; }
    [HalfWidth]
    public int ExamDuration { get; set; }
    [HalfWidth]
    public float MaxMarks { get; set; }
    [HalfWidth]
    public float NegativeMarks { get; set; }
    [HalfWidth]
    public short EExamNavigation { get; set; }
    [HalfWidth]
    public short ESectionNavigation { get; set; }
    [HalfWidth]
    public short EQuestionNavigation { get; set; }
    [HalfWidth]
    public short EResultType { get; set; }
    [HalfWidth]
    public short EOptionDisplayType { get; set; }
    [HalfWidth]
    public bool HasNegativeMarketing { get; set; }
    [HtmlContentEditor(Rows = 4), CssClass("fact")]
    public string Instructions { get; set; }
    [HalfWidth]
    public string SearchTags { get; set; }
    [Category("ExamSectionOption"), CssClass("fact")]
    [ExamSectionEditor]
    public List<ExamSectionRow> ExamSection { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Exams/ExamForm.ts
import { ExamSectionEditor } from "@/Exams/ExamSection/ExamSectionEditor";
import { StringEditor, EnumEditor, ImageUploadEditor, IntegerEditor, DecimalEditor, BooleanEditor, HtmlContentEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EExamState } from "../Web/Enums.EExamState";
import { EExamTypes } from "../Web/Enums.EExamTypes";
import { EOptionDisplayTypes } from "../Web/Enums.EOptionDisplayTypes";
import { EQuestionNavigation } from "../Web/Enums.EQuestionNavigation";
import { EResultTypes } from "../Web/Enums.EResultTypes";

export interface ExamForm {
    Title: StringEditor;
    EExamType: EnumEditor;
    EExamState: EnumEditor;
    QuestionPaperPath: ImageUploadEditor;
    ModelAnswerPaperPath: ImageUploadEditor;
    ExamDuration: IntegerEditor;
    MaxMarks: DecimalEditor;
    NegativeMarks: DecimalEditor;
    EExamNavigation: EnumEditor;
    ESectionNavigation: EnumEditor;
    EQuestionNavigation: EnumEditor;
    EResultType: EnumEditor;
    EOptionDisplayType: EnumEditor;
    HasNegativeMarketing: BooleanEditor;
    Instructions: HtmlContentEditor;
    SearchTags: StringEditor;
    ExamSection: ExamSectionEditor;
}

export class ExamForm extends PrefixedContext {
    static readonly formKey = 'Exams.Exam';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ExamForm.init)  {
            ExamForm.init = true;

            var w0 = StringEditor;
            var w1 = EnumEditor;
            var w2 = ImageUploadEditor;
            var w3 = IntegerEditor;
            var w4 = DecimalEditor;
            var w5 = BooleanEditor;
            var w6 = HtmlContentEditor;
            var w7 = ExamSectionEditor;

            initFormType(ExamForm, [
                'Title', w0,
                'EExamType', w1,
                'EExamState', w1,
                'QuestionPaperPath', w2,
                'ModelAnswerPaperPath', w2,
                'ExamDuration', w3,
                'MaxMarks', w4,
                'NegativeMarks', w4,
                'EExamNavigation', w1,
                'ESectionNavigation', w1,
                'EQuestionNavigation', w1,
                'EResultType', w1,
                'EOptionDisplayType', w1,
                'HasNegativeMarketing', w5,
                'Instructions', w6,
                'SearchTags', w0,
                'ExamSection', w7
            ]);
        }
    }
}

[EExamTypes, EExamState, EQuestionNavigation, EResultTypes, EOptionDisplayTypes]; // referenced types